blocks/bricks/recurrent.py

```diff
@@ -169,17 +172,20 @@ def only_given(arg_names):
                 states_given[name] = tensor.unbroadcast(state,
                                                         *range(state.ndim))
 
-            # Apply methods
-            if not iterate:
-                return application_function(brick, **kwargs)
-
             def scan_function(*args):
                 args = list(args)
                 arg_names = (list(sequences_given) + list(states_given) +
                              list(contexts_given))
                 kwargs = dict(zip(arg_names, args))
                 kwargs.update(rest_kwargs)
-                return application_function(brick, **kwargs)
+                outputs = getattr(brick, application_function.__name__)(
```

Comment:
In fact those are different!

`getattr` calls the decorated version, and the resulting computation graph is properly annotated. I will add an example of how one can use it to the test.